integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/createTable/TestCreateDDLForComplexMapType.scala

```diff
@@ -0,0 +1,330 @@
+    sql("DROP TABLE IF EXISTS carbon")
+    sql(
+      s"""
+         | CREATE TABLE carbon(
+         | mapField map<STRING,map<INT,STRING>>
+         | )
+         | STORED BY
+         |'carbondata' """
+        .stripMargin)
+    sql(
+      "insert into carbon values('manish:1\004nalla\0032\004gupta$kunal:1\004kapoor\0032\004sharma')")
+    checkAnswer(sql("select * from carbon"), Seq(
+      Row(Map("manish" -> Map(1 -> "nalla", 2 -> "gupta"),
+        "kunal" -> Map(1 -> "kapoor", 2 -> "sharma")))))
+  }
+
+  test("Test Load duplicate keys data in map of map") {
+    sql("DROP TABLE IF EXISTS carbon")
+    sql(
+      s"""
+         | CREATE TABLE carbon(
+         | mapField map<STRING,map<INT,STRING>>
+         | )
+         | STORED BY
+         |'carbondata'
+         |"""
+        .stripMargin)
+    sql(
+      "insert into carbon values('manish:1\004nalla\0031\004gupta$kunal:1\004kapoor\0032\004sharma')")
+    checkAnswer(sql("select * from carbon"), Seq(
+      Row(Map("manish" -> Map(1 -> "nalla"),
+        "kunal" -> Map(1 -> "kapoor", 2 -> "sharma")))))
+  }
+
+  test("Test Create table as select with map") {
+    sql("DROP TABLE IF EXISTS carbon")
+    sql("DROP TABLE IF EXISTS carbon1")
+    sql(
+      s"""
+         | CREATE TABLE carbon(
+         | mapField map<INT,STRING>
+         | )
+         | STORED BY 'carbondata'
+         | """
+        .stripMargin)
+    sql("insert into carbon values('1:Nalla$2:Singh$3:Gupta$4:Kumar')")
+    sql(
+      s"""
+         | CREATE TABLE carbon1
+         | AS
+         | Select *
+         | From carbon
+         | """
+        .stripMargin)
+    checkAnswer(sql("select * from carbon1"), Seq(
+      Row(Map(1 -> "Nalla", 2 -> "Singh", 3 -> "Gupta", 4 -> "Kumar"))))
+  }
+
+  test("Test Create table with double datatype in map") {
+    sql("DROP TABLE IF EXISTS carbon")
+    sql(
+      s"""
+         | CREATE TABLE carbon(
+         | mapField map<DOUBLE,STRING>
+         | )
+         | STORED BY 'carbondata'
+         | """
+        .stripMargin)
+    sql("insert into carbon values('1.23:Nalla$2.34:Singh$3.67676:Gupta$3.67676:Kumar')")
+    checkAnswer(sql("select * from carbon"), Seq(
+      Row(Map(1.23 -> "Nalla", 2.34 -> "Singh", 3.67676 -> "Gupta"))))
+  }
+
+  test("Load Map data from CSV File") {
+    sql("DROP TABLE IF EXISTS carbon")
+    sql(
+      s"""
+         | CREATE TABLE carbon(
+         | mapField map<INT,STRING>
+         | )
+         | STORED BY 'carbondata'
+         | """
+        .stripMargin)
+    sql(
+      s"""
+         | LOAD DATA LOCAL INPATH '$path'
+         | INTO TABLE carbon
+         | OPTIONS('header'='false')
+       """.stripMargin)
+    checkAnswer(sql("select * from carbon"), Seq(
+      Row(Map(1 -> "Nalla", 2 -> "Singh", 4 -> "Kumar")),
+      Row(Map(10 -> "Nallaa", 20 -> "Sissngh", 100 -> "Gusspta", 40 -> "Kumar"))
+    ))
+  }
+
+  test("Sort Column table property blocking for Map type") {
+    sql("DROP TABLE IF EXISTS carbon")
```

Comment:
Can you please add some testcases for compaction, partition, preaggregate, IUD and alter scenarios